src/openms_gui/include/OpenMS/VISUAL/DIALOGS/SwathTabWidget.h

```diff
@@ -46,23 +47,53 @@ namespace Ui
 
 namespace OpenMS
 {
+  class InputFile;
+  class OutputDirectory;
+  class ParamEditor;
+
   namespace Internal
   {
     /// A multi-tabbed widget for the SwathWizard offering setting of parameters, input-file specification and running Swath and more
     class OPENMS_GUI_DLLAPI SwathTabWidget : public QTabWidget
     {
-        Q_OBJECT
+      Q_OBJECT
 
     public:
-        explicit SwathTabWidget(QWidget *parent = nullptr);
-        ~SwathTabWidget();
+      explicit SwathTabWidget(QWidget *parent = nullptr);
+      ~SwathTabWidget();
     
     private slots:
-      void on_pushButton_clicked();
+      void on_run_swath_clicked();
+      void on_edit_advanced_parameters_clicked();
+      /// update the current working directory for all file input fields
+      void broadcastNewCWD_(const QString& new_cwd);
 
     private:
-        Ui::SwathTabWidget *ui;
+
+      /// collect all parameters throughout the Wizard's controls and update 'swath_param_'
+      void updateSwathParamFromWidgets_();
+
+      /// update Widgets given a param object
+      void updateWidgetsfromSwathParam_();
+
+      /// append text to the log tab
+      /// @param text The text to write
+      /// @param new_section Start a new block with a date and time
+      void writeLog_(const QString& text, bool new_section = false);
+
+      /// Ensure all input widgets are filled with data by the user
+      /// If anything is missing: show a Messagebox and return false.
+      bool checkInputReady_();
+
+      Ui::SwathTabWidget *ui;
+      Param swath_param_; ///< the global Swath parameters which will be passed to OpenSwathWorkflow.exe, once updated with parameters the Wizard holds separately
+      Param swath_param_wizard_; ///< small selection of important parameters which the user can directly change in the Wizard
     };
 
   }
-}
+} // ns OpenMS
+
+// this is required to allow Ui_SwathTabWidget (auto UIC'd from .ui) to have a InputFile member
+using InputFile = OpenMS::InputFile;
```

Comment:
would it make sense to put them in an extra GUI or VISUAL namespace?

Reply:
Ui ?

Reply:
the problem is the auto-generated header from the .ui file: it references an `InputFile* obj;`. 
We'd need to convince the UIC to generate `OpenMS::Inputfile* obj` or similar, but I did not bother to look for how to do that (yet)